### ISSUE TITLE
Fails to apply configuration on Tmux3.0a

### ISSUE THREAD

**Issue by DrMagPie** [gpakosz/.tmux#330]
Hi,

I'm running latest version of Oh My TMUX
```
~/.tmux # git pull
Already up to date.
~/.tmux #
```
OS: Alpine Linux on WSL
Terminal Emulator: Alacritty

everything works on  tmux 2.9 
but on tmux3.0a `cut -c3- ~/.tmux.conf | sh -s _apply_configuration` returns error code 255

Execution log:
```
~/.tmux # cut -c3- ~/.tmux.conf | sh -sx _apply_configuration
+ cat
+ set -e
+ unset GREP_OPTIONS
+ export 'LC_NUMERIC=C'
+ printf
+ sed -E s///
+ __newline='
'
+ _apply_configuration
+ command -v perl
+ command -v sed
+ command -v awk
+ command -v reattach-to-user-namespace
+ _apply_overrides
+ tmux_conf_theme_24b_colour=false
+ _is_enabled false
+ '[' xfalse '=' xenabled ]
+ '[' xfalse '=' xtrue ]
+ '[' xfalse '=' xyes ]
+ '[' xfalse '=' x1 ]
+ return 1
+ _apply_bindings
+ mktemp
+ cfg=/tmp/tmp.8SPWb46nJm
+ trap 'rm -f $cfg*' EXIT
+ tmux list-keys
+ grep -vF tmux.conf.local
+ grep -E '(new-window|split(-|_)window|new-session|copy-selection|copy-pipe)'
+ perl -p -i -e 's/'"'"'#\{\?window_zoomed_flag,Unzoom,Zoom\}'"'"' '"'"'z'"'"' \{resize-pane -$/'"'"'#{?wi
ndow_zoomed_flag,Unzoom,Zoom}'"'"' '"'"'z'"'"' {resize-pane -Z}"/g' /tmp/tmp.8SPWb46nJm
+ tmux_conf_new_window_retain_current_path=false
+ _is_enabled false
+ '[' xfalse '=' xenabled ]
+ '[' xfalse '=' xtrue ]
+ '[' xfalse '=' xyes ]
+ '[' xfalse '=' x1 ]
+ return 1
+ perl -p -i -e 's/\bnew-window\b([^;}\n"]*?)(?:\s+-c\s+(\\?"?|'"'"'?)#\{pane_current_path\}\2)/new-window
\1/g' /tmp/tmp.8SPWb46nJm
+ tmux_conf_new_pane_retain_current_path=true
+ _is_enabled true
+ '[' xtrue '=' xenabled ]
+ '[' xtrue '=' xtrue ]
+ return 0
+ perl -p -i -e 's/\bsplit-window\b(?!([^;}\n"]*?)(?:\s+-c\s+(\\?"?|'"'"'?)#\{pane_current_path\}\2))/spli
t-window -c '"'"'#{pane_current_path}'"'"'/g' /tmp/tmp.8SPWb46nJm
+ tmux_conf_new_pane_reconnect_ssh=false
+ _is_enabled false
+ '[' xfalse '=' xenabled ]
+ '[' xfalse '=' xtrue ]
+ '[' xfalse '=' xyes ]
+ '[' xfalse '=' x1 ]
+ return 1
+ _is_enabled true
+ '[' xtrue '=' xenabled ]
+ '[' xtrue '=' xtrue ]
+ return 0
+ perl -p -i -e 's/\brun-shell\b(\s+("|'"'"')cut\s+-c3-\s+~\/\.tmux\.conf\s+\|\s+sh\s+-s\s+_split_window\s
+#\{pane_tty\})(.*?)\s+-c\s+#\{pane_current_path\}\2/split-window\3 -c '"'"'#{pane_current_path}'"'"'/g' /
tmp/tmp.8SPWb46nJm
+ tmux_conf_new_session_prompt=false
+ _is_enabled false
+ '[' xfalse '=' xenabled ]
+ '[' xfalse '=' xtrue ]
+ '[' xfalse '=' xyes ]
+ '[' xfalse '=' x1 ]
+ return 1
+ perl -p -i -e 's/\bcommand-prompt\s+-p\s+new-session\s+'"'"'new-session\s+-s\s+"%%"'"'"'/new-session/g'
/tmp/tmp.8SPWb46nJm
+ tmux_conf_copy_to_os_clipboard=false
+ command -v pbcopy
+ command -v reattach-to-user-namespace
+ command -v xsel
+ command -v xsel
+ command -v xclip
+ command -v clip.exe
+ '[' -c /dev/clipboard ]
+ '[' -n  ]
+ awk '{i = $2 == "-T" ? 4 : 5; gsub(/^[;]$/, "\\\\&", $i); gsub(/^[$"#~]$/, "'"'"'&'"'"'", $i); gsub(/^['
"'"']$/, "\"&\"", $i); print}'
+ tmux source-file /tmp/tmp.8SPWb46nJm.in
usage: if-shell [-bF] [-t target-pane] shell-command command [command]
+ out=
+ printf '%s'
+ cut -d: -f2
+ line=
+ perl -n -i -e 'if ($. != ) { print }' /tmp/tmp.8SPWb46nJm.in
syntax error at -e line 1, near "!= ) "
syntax error at -e line 1, near ";}"
Execution of -e aborted due to compilation errors.
+ rm -f /tmp/tmp.8SPWb46nJm /tmp/tmp.8SPWb46nJm.in
```

It seams that issue is [here](https://github.com/gpakosz/.tmux/blob/master/.tmux.conf#L758-L761)
For some reason tmux3 doesn't return output 
and line variable is empty so perl crashes

**Comment by gpakosz** [comment]
Hello @DrMagPie 👋

Can you please paste the full output of
```
$ tmux list-keys | grep -vF 'tmux.conf.local' | grep -E '(new-window|split(-|_)window|new-session|copy-selection|copy-pipe)'
```

Thank you

**Comment by DrMagPie** [comment]
Hi, 
here is out put:
```
bind-key    -T copy-mode    C-w                   send-keys -X \230*y\213O\177 copy-selection-and-cancel
bind-key    -T copy-mode    MouseDragEnd1Pane     send-keys -X \230*y\213O\177 copy-selection-and-cancel
bind-key    -T copy-mode    M-w                   send-keys -X \230*y\213O\177 copy-selection-and-cancel
bind-key    -T copy-mode-vi C-j                   send-keys -X copy-selection-and-cancel copy-selection-and-cancel
bind-key    -T copy-mode-vi Enter                 send-keys -X copy-selection-and-cancel copy-selection-and-cancel
bind-key    -T copy-mode-vi y                     send-keys -X copy-selection-and-cancel copy-mode-vi copy-selection-and-cancel
bind-key    -T copy-mode-vi MouseDragEnd1Pane     send-keys -X copy-selection-and-cancel copy-selection-and-cancel
bind-key    -T prefix       C-c                   new-session
bind-key    -T prefix       \"                    split-window -c "#{pane_current_path}"
bind-key    -T prefix       %                     split-window -c "#{pane_current_path}" -h "#{pane_current_path}" \331\302\377\177
bind-key    -T prefix       -                     split-window -c "#{pane_current_path}" -v "#{pane_current_path}" split-window -c "#{pane_current_path}"
bind-key    -T prefix       _                     split-window -c "#{pane_current_path}" -h "#{pane_current_path}"
bind-key    -T prefix       c                     new-window
bind-key    -T root         MouseDown3Pane        if-shell -F "#[align=centre]#{window_index}:#{window_name}" -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -h} 'Vertical Split' 'v' {split-window -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill' 'X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
bind-key    -T root         MouseDown3Status      display-menu -T "#[align=centre]#{window_index}:#{window_name}" -t = -x W -y S "Swap Left" l "swap-window -t:-1" "Swap Right" r "swap-window -t:+1" "#{?pane_marked_set,,-}Swap Marked" s swap-window Kill X kill-window Respawn R "respawn-window -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" Rename n "command-prompt -I \"#W\" \"rename-window -- '%%'\"" "New After" w "new-window -a" "New At End" W new-window
bind-key    -T root         MouseDown3StatusLeft  display-menu -T "#[align=centre]#{session_name}" -t = -x M -y S Next n "switch-client -n" Previous p "switch-client -p" Renumber N "move-window -r" Rename n "command-prompt -I \"#S\" \"rename-session -- '%%'\"" "New Session" s new-session "New Window" w new-window
bind-key    -T root         M-MouseDown3Pane      display-menu -T "#[align=centre]#{pane_index} (#{pane_id})" -t = -x M -y M "#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}" C-r "copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"" "#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}" C-y "send-keys -l -- \"#{q:mouse_word}\"" "#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}" c "set-buffer -- \"#{q:mouse_word}\"" "#{?mouse_line,Copy Line,}" l "set-buffer -- \"#{q:mouse_line}\"" "Horizontal Split" h "split-window -c '#{pane_current_path}' -h" "Vertical Split" v "split-window -c '#{pane_current_path}' -v" "Swap Up" u "swap-pane -U" "Swap Down" d "swap-pane -D" "#{?pane_marked_set,,-}Swap Marked" s swap-pane Kill X kill-pane Respawn R "respawn-pane -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" "#{?window_zoomed_flag,Unzoom,Zoom}" z "resize-pane -Z"
```

**Comment by gpakosz** [comment]
Can you please do the same with
```
$ tmux -L test -f /dev/null list-keys | grep -vF 'tmux.conf.local' | grep -E '(new-window|split(-|_)window|new-session|copy-selection|copy-pipe)'
```

**Comment by DrMagPie** [comment]
```
bind-key    -T copy-mode    C-w                   send-keys -X  copy-selection-and-cancel
bind-key    -T copy-mode    MouseDragEnd1Pane     send-keys -X \236g\177 copy-selection-and-cancel
bind-key    -T copy-mode    M-w                   send-keys -X  copy-selection-and-cancel
bind-key    -T copy-mode-vi C-j                   send-keys -X  copy-selection-and-cancel
bind-key    -T copy-mode-vi Enter                 send-keys -X  copy-selection-and-cancel
bind-key    -T copy-mode-vi MouseDragEnd1Pane     send-keys -X \356\377\177 copy-selection-and-cancel
bind-key    -T prefix       \"                    split-window
bind-key    -T prefix       %                     split-window -h 
bind-key    -T prefix       <                     display-menu -T "#[align=centre]#{window_index}:#{window_name}" -x W -y S "Swap Left" l "swap-window -t:-1" "Swap Right" r "swap-window -t:+1" "#{?pane_marked_set,,-}Swap Marked" s swap-window  Kill X kill-window Respawn R "respawn-window -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" Rename n "command-prompt -I \"#W\" \"rename-window -- '%%'\""  "New After" w "new-window -a" "New At End" W new-window
bind-key    -T prefix       >                     display-menu -T "#[align=centre]#{pane_index} (#{pane_id})" -x P -y P "#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}" C-r "copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"" "#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}" C-y "send-keys -l -- \"#{q:mouse_word}\"" "#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}" c "set-buffer -- \"#{q:mouse_word}\"" "#{?mouse_line,Copy Line,}" l "set-buffer -- \"#{q:mouse_line}\""  "Horizontal Split" h "split-window -h" "Vertical Split" v "split-window -v"  "Swap Up" u "swap-pane -U" "Swap Down" d "swap-pane -D" "#{?pane_marked_set,,-}Swap Marked" s swap-pane  Kill X kill-pane Respawn R "respawn-pane -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" "#{?window_zoomed_flag,Unzoom,Zoom}" z "resize-pane -Z"
bind-key    -T prefix       c                     new-window
bind-key    -T root         MouseDown3Pane        if-shell -F "#[align=centre]#{window_index}:#{window_name}" -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -h} 'Vertical Split' 'v' {split-window -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill' 'X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
bind-key    -T root         MouseDown3Status      display-menu -T "#[align=centre]#{window_index}:#{window_name}" -t = -x W -y S "Swap Left" l "swap-window -t:-1" "Swap Right" r "swap-window -t:+1" "#{?pane_marked_set,,-}Swap Marked" s swap-window  Kill X kill-window Respawn R "respawn-window -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" Rename n "command-prompt -I \"#W\" \"rename-window -- '%%'\""  "New After" w "new-window -a" "New At End" W new-window
bind-key    -T root         MouseDown3StatusLeft  display-menu -T "#[align=centre]#{session_name}" -t = -x M -y S Next n "switch-client -n" Previous p "switch-client -p"  Renumber N "move-window -r" Rename n "command-prompt -I \"#S\" \"rename-session -- '%%'\""  "New Session" s new-session "New Window" w new-window
bind-key    -T root         M-MouseDown3Pane      display-menu -T "#[align=centre]#{pane_index} (#{pane_id})" -t = -x M -y M "#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}" C-r "copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"" "#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}" C-y "send-keys -l -- \"#{q:mouse_word}\"" "#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}" c "set-buffer -- \"#{q:mouse_word}\"" "#{?mouse_line,Copy Line,}" l "set-buffer -- \"#{q:mouse_line}\""  "Horizontal Split" h "split-window -h" "Vertical Split" v "split-window -v"  "Swap Up" u "swap-pane -U" "Swap Down" d "swap-pane -D" "#{?pane_marked_set,,-}Swap Marked" s swap-pane  Kill X kill-pane Respawn R "respawn-pane -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" "#{?window_zoomed_flag,Unzoom,Zoom}" z "resize-pane -Z"
```

**Comment by gpakosz** [comment]
Thank you.

I'm surprised to say the least. `-L test` is supposed to make sure another server is started. `-f /dev/null` avoids loading any configuration file. In the output you pasted, it seems the following line
```
bind-key    -T root         MouseDown3Pane        if-shell -F "#[align=centre]#{window_index}:#{window_name}" -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -h} 'Vertical Split' 'v' {split-window -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill' 'X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
```
can't be read back by tmux.

As I can't reproduce the same output as you, that is the `MouseDown3Pane        if-shell -F "#[align=centre]#{window_index}:#{window_name}" ...` part, can you please reboot and before launching any tmux process, do again:
```
$ tmux -L test -f /dev/null list-keys | grep MouseDown3Pane
```
and paste the output?

Thank you

**Comment by DrMagPie** [comment]
```
bind-key    -T root         MouseDown3Pane        if-shell -F \360,\023\r\316\177 -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -h} 'Vertical Split' 'v' {split-window -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill' 'X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
bind-key    -T root         M-MouseDown3Pane      display-menu -T "#[align=centre]#{pane_index} (#{pane_id})" -t = -x M -y M "#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}" C-r "copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"" "#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}" C-y "send-keys -l -- \"#{q:mouse_word}\"" "#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}" c "set-buffer -- \"#{q:mouse_word}\"" "#{?mouse_line,Copy Line,}" l "set-buffer -- \"#{q:mouse_line}\""  "Horizontal Split" h "split-window -h" "Vertical Split" v "split-window -v"  "Swap Up" u "swap-pane -U" "Swap Down" d "swap-pane -D" "#{?pane_marked_set,,-}Swap Marked" s swap-pane  Kill X kill-pane Respawn R "respawn-pane -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" "#{?window_zoomed_flag,Unzoom,Zoom}" z "resize-pane -Z"
```

**Comment by DrMagPie** [comment]
out of curiosity I have run the same command in tmux2.9a-r1, and here is output:
```
bind-key    -T root         MouseDown3Pane    if-shell -F -t = "#{mouse_any_flag}" "select-pane -t=; send-keys -M" "select-pane -mt="
```

**Comment by gpakosz** [comment]
I'm very surprised by
```
bind-key    -T root         MouseDown3Pane        if-shell -F \360,\023\r\316\177 -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -h} 'Vertical Split' 'v' {split-window -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill' 'X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
```

If you put that line in e.g. `/tmp/tmux.conf` and then do `$ tmux source /tmp/tmux.conf`, does that work?

**Comment by DrMagPie** [comment]
```
$ vim /tmp/tmux.conf
$ cat /tmp/tmux.conf
bind-key    -T root         MouseDown3Pane        if-shell -F \360,\023\r\316\177 -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -h} 'Vertical Split' 'v' {split-window -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill''X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
$ tmux source /tmp/tmux.conf
usage: if-shell [-bF] [-t target-pane] shell-command command [command]
```

**Comment by gpakosz** [comment]
Right so it seems your version of tmux can't `source` the output of `list-keys`. Which is not supposed to happen with tmux >= 3.0 as far as I know.

I can mitigate things on my side but then not all bindings will work. That's definitely a tmux bug.

Out of curiosity, can you please tell me what OS you're using and how you installed tmux?

**Comment by DrMagPie** [comment]
Sure,
I'm Running Alpine linux on WSL 1 (From windows store)
tmux is installed using apk package manager `apk add tmux`
and here is my repo list
```
cat /etc/apk/repositories
http://dl-cdn.alpinelinux.org/alpine/edge/main
http://dl-cdn.alpinelinux.org/alpine/edge/community
```

P.S.
tmux2.9 also can't source this line, but during configuration this line does not appear, so everything works on 2.9
```
bind-key    -T root         MouseDown3Pane        if-shell -F \360,\023\r\316\177 -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -h} 'Vertical Split' 'v' {split-window -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill''X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
```
also I have tested it on Mac OS with tmux3.1a and everything works as expected.
so looks like only Alpine version if affected by this bug.

**Comment by DrMagPie** [comment]
Alpine linux is not really popular as "`desktop`" distro, so I'm not sure if it's needed to be fixed. especially if it breaks some stuff on your end. :(

**Comment by gpakosz** [comment]
Here's what I have on macOS Catalina
```
$ tmux -V
tmux 3.0a
$ tmux -L test -f /dev/null list-keys | grep MouseDown3Pane
bind-key    -T root         MouseDown3Pane        if-shell -F -t = "#{||:#{mouse_any_flag},#{pane_in_mode}}" "select-pane -t=; send-keys -M" "display-menu -t= -xM -yM -T \"#[align=centre]#{pane_index} (#{pane_id})\"  '#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}' 'C-r' {copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"} '#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}' 'C-y' {send-keys -l -- \"#{q:mouse_word}\"} '#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}' 'c' {set-buffer -- \"#{q:mouse_word}\"} '#{?mouse_line,Copy Line,}' 'l' {set-buffer -- \"#{q:mouse_line}\"} '' 'Horizontal Split' 'h' {split-window -c '#{pane_current_path}' -h} 'Vertical Split' 'v' {split-window -c '#{pane_current_path}' -v} '' 'Swap Up' 'u' {swap-pane -U} 'Swap Down' 'd' {swap-pane -D} '#{?pane_marked_set,,-}Swap Marked' 's' {swap-pane} '' 'Kill' 'X' {kill-pane} 'Respawn' 'R' {respawn-pane -k} '#{?pane_marked,Unmark,Mark}' 'm' {select-pane -m} '#{?window_zoomed_flag,Unzoom,Zoom}' 'z' {resize-pane -Z}"
bind-key    -T root         M-MouseDown3Pane      display-menu -T "#[align=centre]#{pane_index} (#{pane_id})" -t = -x M -y M "#{?mouse_word,Search For #[underscore]#{=/9/...:mouse_word},}" C-r "copy-mode -t=; send -Xt= search-backward \"#{q:mouse_word}\"" "#{?mouse_word,Type #[underscore]#{=/9/...:mouse_word},}" C-y "send-keys -l -- \"#{q:mouse_word}\"" "#{?mouse_word,Copy #[underscore]#{=/9/...:mouse_word},}" c "set-buffer -- \"#{q:mouse_word}\"" "#{?mouse_line,Copy Line,}" l "set-buffer -- \"#{q:mouse_line}\"" "Horizontal Split" h "split-window -c '#{pane_current_path}' -h" "Vertical Split" v "split-window -c '#{pane_current_path}' -v" "Swap Up" u "swap-pane -U" "Swap Down" d "swap-pane -D" "#{?pane_marked_set,,-}Swap Marked" s swap-pane Kill X kill-pane Respawn R "respawn-pane -k" "#{?pane_marked,Unmark,Mark}" m "select-pane -m" "#{?window_zoomed_flag,Unzoom,Zoom}" z "resize-pane -Z"
```

**Comment by gpakosz** [comment]
I suspect there's a problem with the way tmux is compiled for Alpine Linux or maybe there's a problem with your locale

**Comment by gpakosz** [comment]
You may want to report the issue to tmux authors

**Comment by DrMagPie** [comment]
looked seams to be okay.
```
$ locale
LANG=C.UTF-8
LC_CTYPE=C.UTF-8
LC_NUMERIC=C.UTF-8
LC_TIME=C.UTF-8
LC_COLLATE=C.UTF-8
LC_MONETARY=C.UTF-8
LC_MESSAGES=C.UTF-8
LC_ALL=
```
although quick googling showed that Alpine (or better to say musl library) has some issues with locale.

so it's probably worth reporting to tmux team.
Thank you for help 😊

**Comment by gpakosz** [comment]
![image](https://user-images.githubusercontent.com/553208/76861359-c1726100-685c-11ea-99b1-8187b83d399e.png)

From Alpine for WSL freshly installed 🤷‍♂️

**Comment by gpakosz** [comment]
Confirmed bug with tmux authors. I'll try to workaround it but it means some bindings won't benefit from  settings like `tmux_conf_new_window_retain_current_path`, `tmux_conf_new_pane_retain_current_path`, ...

**Comment by gpakosz** [comment]
Fixed in upstream: tmux/tmux@bd0342b0a7ac4b5dc00ffc9cce7f9b0fbeb7ec91

**Comment by gpakosz** [comment]
@DrMagPie can you please test the `gh-330` branch? This will make `_apply_bindings()` properly ignore incorrect bindings.

Note that I can't fix the root cause of tmux 3.0a being broken on Alpine because it assumes `getopt` resets `optarg` to `NULL` when options have no arguments.

As such, while `_apply_bindings()` will succeed, `MouseDown3Pane` binding won't benefit from `tmux_conf_new_pane_retain_current_path=true`

**Comment by DrMagPie** [comment]
works! :)
![image](https://user-images.githubusercontent.com/23652900/76892107-cce07f00-6892-11ea-9408-0cc46d6542d6.png)
